src/opencl/mod.rs

```diff
@@ -114,14 +118,15 @@ impl Device {
     pub fn is_little_endian(&self) -> GPUResult<bool> {
         Ok(utils::is_little_endian(self.device)?)
     }
-    pub fn bus_id(&self) -> BusId {
+    pub fn bus_id(&self) -> Option<BusId> {
         self.bus_id
     }
 
+    // Return all available GPU devices of supported brands, ordered by brand as
+    // defined by Brand::all().
     pub fn all() -> GPUResult<Vec<Device>> {
         let mut all = Vec::new();
-
-        for b in &[Brand::Nvidia, Brand::Amd] {
+        for b in &Brand::all() {
```

Comment:
Why do we bother fitlering by brand, if we really want all platforms with specific devices?

Reply:
So the result will be returned in priority order by brand.